### PR TITLE
Chrisv/scrolling

### PR DESCRIPTION
**Pull request by biovisualize**
* scrolling panel for all charts
* `useScrolling` option
* works with a min mark width, doesn't scroll if everything fits
* better handling of computed size (autoconfig.js resolve "auto" options and pushes chartWidth and such down to components)
* 2nd axis works
* minor cleanup (`.mapd3` namespace, syntax, unused code)

### CODE REVIEW

File: src/charts/chart.js [hunk 1]
```diff
@@ -398,7 +418,10 @@ export default function Chart (_container) {
   }
 
   function setConfig (_config) {
-    config = override(config, _config)
+    inputConfig = override(inputConfig, _config)
+
+    const autoConfig = autoConfigure(inputConfig, cache, data)
```
Comment (clhenrick) [inline src/charts/chart.js]: "data" is not defined within the scope of this function or scope of the module

**Comment by biovisualize** [comment]
Notes from conversation with @clhenrick:
* data -> dataObject
* try it in Immerse. Their might be some confusion with data/config and getting sizes from the DOM.

**Comment by clhenrick** [comment]
We probably should remove the left margin from `transfrom: translate(x, y)` when `useScroll` is enabled:

screen shot 2018-03-29 at 1 38 34 pm

**Comment by clhenrick** [comment]
@biovisualize I'm not following you here:

> No, I think it's because you are setting the left margin to 0

So setting the chart `config.margin.left` to `0` would result in the chart rendering with `margin.left = 70` ?

**Comment by clhenrick** [comment]
Just noticing that the tooltip doesn't show if you mouse outside of the bars:

mapd3-scrolling-tooltip-bug

**Comment by biovisualize** [comment]
@clhenrick no, config.margin.left to 0 will set it to 0. Mmh, there's something wrong with your setup. https://github.com/mapd/mapd-immerse/blob/master/src/charts/bar-chart/bar-adapter.js#L80 is supposed to be set to 70 and not show the padding you mention. Let's look at it together.

**Comment by biovisualize** [comment]
Fixed https://github.com/mapd/mapd3/pull/27#issuecomment-377404870

**Comment by clhenrick** [comment]
@biovisualize you're correct, when pointing to the correct branch the `margin.left` works as expected. Thanks for fixing the tooltip panel. 

Is there anything else that needs to be done before merging this PR?

**Comment by biovisualize** [comment]
I don't think so. I'm using it on line2 and it works well so far. I will merge it and you can point your Immerse branch to tip of master once done.

**Pull request merged**